### ISSUE TITLE
Use alternate status code for warnings

### ISSUE THREAD

**Issue by duckbrain** [agent-ecosystem/skill-validator#6]
Currently the status codes are:

- Pass - 0
- Warning - 0
- Error - 1

This means you can't tell if there are warnings based on the status code for CI pipelines. If the warning status could be changed eg: 2.  Then pipelines could be set up to warn on those statuses.

Workaround with a wrapper script:

```shell
#!/bin/sh

output=$(skill-validator "$@")
status=$?
echo "$output"
if [ "$status" -ne 0 ]; then
    exit 1
fi
if echo "$output" | tail -n 2 | grep -qi "warning"; then
    exit 2
fi
exit 0
```

Then I can set up a job to warn on warnings in GitLab CI

```
validate:
  image: golang:1.25-alpine3.22
  stage: validate
  allow_failure:
    exit_codes:
      - 2
  script: ...
```

**Comment by dacharyc** [comment]
Thanks for opening an issue @duckbrain ! I'm setting up the tool in CI myself right now and am considering what functionality might be helpful for my case.

For your case, would a `--strict` flag be sufficient? I've got that captured in proposed enhancement: #11 

I'm also considering this as part of a handful of related work:
- #13 
- #14 
- #12 

Would those help with how you're picturing setting up your CI? It looks like you're using GitLab and I'm not sure these enhancements would map directly to CI set up in GitLab, or if your needs are different.

**Comment by duckbrain** [comment]
Thanks! Works great.
